public/client.js

```diff
@@ -1,108 +1,158 @@
-      });
-      $('.trucks-list').append(list);
-    }
-  });
-
-  $('form').on('submit', function (e) {
-
-    e.preventDefault();
-    var $form = $(this);
-
-    var truckData = {
-      name: $('[name=name]').val(),
-      type: foodTypes,
-      schedule: getSchedule(),
-      description: $('[name=description]').val(),
-      payment: getPaymentTypes(),
-      website: $('[name=website]').val(),
-      Facebook: $('[name=Facebook]').val(),
-      Twitter: $('[name=Twitter]').val()
-    };
-
-    $.ajax({
-      method: 'POST',
-      url: '/trucks',
-      data: truckData
-    })
-    .done(function (truck) {
-      var item = '<li><span class="delete_link" data-truck="' + truck.name + '">X</span><a href="/trucks/' + truck.name + '">' + truck.name + '</a></li>';
-
-      $('.trucks-list').append(item);
-      $form.trigger('reset');
-    });
-  });
-
-  function getPaymentTypes() {
-    var types = [];
-
-    $('[name=payment]').each(function () {
-      if (this.checked) {
-        types.push(this.value);
-      }
-    });
-
-    return types;
-  }
-
-  function getSchedule() {
-    var schedule = [];
-
-    $('[name=schedule]').each(function () {
-      if (this.checked) {
-        schedule.push(this.value);
-      }
-    });
-
-    return schedule;
-  }
-
-  function addFoodType(type) {
-
-    foodTypes.push(type);
-    $('.foodType-list').append('<li>' + type + '</li>');
-    $('[name=type]').val('');
-  }
-
-  $('[name=type]').on('keypress', function (e) {
-    if (e.which === 13) {
-      e.preventDefault();
-      addFoodType($(this).val());
-    }
-  });
-
-  $('#addFoodType').on('click', function (e) {
-    var foodType = $('[name=type]').val();
-
-    addFoodType(foodType);
-  });
-
-  $('#clearFoodTypes').on('click', function (e) {
-    $('.foodType-list').empty();
-  });
-
-  $('.trucks-list').on('click', '[data-truck]', function (e) {
-    if (!confirm('Remove food truck?')) {
-      return false;
-    }
-    var $target = $(e.currentTarget);
-
-    $.ajax({
-      method: 'DELETE',
-      url: '/trucks/' + $target.data('truck'),
-    })
-    .done(function () {
-      $target.closest('li').remove();
-    });
-  });
-});
\ No newline at end of file
+	"use strict";
+
+	var foodTruckOptions = [];
+	var foodTypes = [];
+
+	$.get("/trucks", function (data) {
+		var list = [];
+
+		if (data && Array.isArray(data)) {
+			data.forEach(function (truck) {
+				foodTruckOptions.push("<option value=\"" + truck.name + "\">" + truck.name + "</option>");
+				list.push("<li><li><span class=\"delete_link\" data-truck=\"" + truck.name + "\">X</span><a href=\"/trucks/" + truck.name + "\">" + truck.name + "</a></li>");
+			});
+			$(".trucks-list").append(list);
+		}
+	});
+
+	$("form").on("submit", function (e) {
+		e.preventDefault();
+		var $form = $(this);
+
+		// serialize will transform our form data into urlencoded notation
+		var truckData = $form.serialize();
+
+		if ($("input[type=radio][name=update]").value === "new") {
```

Comment:
This is close - the problem is the selector you're using - instead use `"input[name=update]:checked"` for both conditionals.